src/main/java/com/microsoft/sqlserver/jdbc/SQLServerSpatialDatatype.java

```
@@ -184,23 +203,544 @@ protected void serializeToWkb(boolean excludeZMFromWKB, SQLServerSpatialDatatype
+         * 000000000000F03F0000000000000040 16 bytes, 2 points with x and y axis (1, 2)
+         * 00000000000008400000000000000040 16 bytes, 2 points with x and y axis (3, 2)
+         * 
+         * There are geometric objects that contain other geometric objects, such as MULTIPOINT.
+         * The below logic builds WKB from existing Geometry/Geography object and returns a byte array.
+         */
+
+        buf.order(ByteOrder.LITTLE_ENDIAN);
+        switch (internalType) {
+            case POINT:
+                addPointToBuffer(buf, numberOfPoints);
+                break;
+            case LINESTRING:
+                addLineStringToBuffer(buf, numberOfPoints);
+                break;
+            case POLYGON:
+                addPolygonToBuffer(buf, numberOfFigures);
+                break;
+            case MULTIPOINT:
+                addMultiPointToBuffer(buf, numberOfFigures);
+                break;
+            case MULTILINESTRING:
+                addMultiLineStringToBuffer(buf, numberOfFigures);
+                break;
+            case MULTIPOLYGON:
+                addMultiPolygonToBuffer(buf, numberOfShapes - 1);
+                break;
+            case GEOMETRYCOLLECTION:
+                addGeometryCollectionToBuffer(buf, calculateNumShapesInThisGeometryCollection());
+                break;
+            case CIRCULARSTRING:
+                addCircularStringToBuffer(buf, numberOfPoints);
+                break;
+            case COMPOUNDCURVE:
+                addCompoundCurveToBuffer(buf, calculateNumCurvesInThisFigure());
+                break;
+            case CURVEPOLYGON:
+                addCurvePolygonToBuffer(buf, numberOfFigures);
+                break;
+            case FULLGLOBE:
+                addFullGlobeToBuffer(buf);
+                break;
+            default:
+                // Shouldn't be possible to come here, existing Geometry/Geography doesn't support other cases
+                break;
```

Need to throw an exception here instead.